ethstorage/downloader/blob_disk_cache.go

```diff
@@ -58,12 +62,18 @@ func (c *BlobDiskCache) SetBlockBlobs(block *blockBlobs) error {
 
 	var blbs []*blob
 	for _, b := range block.blobs {
+		kvi := b.kvIndex.Uint64()
+		if id, ok := c.kvIndexLookup[kvi]; ok {
+			if err := c.store.Delete(id); err != nil {
```

Comment:
This is not the situation I mentioned. What I referred to above concerns the re-org case. Actually, this change will cause an error because the earlier block that first updates the key-value will not retrieve its data back.